documentation/openapi.json

```diff
@@ -754,6 +754,36 @@
         }
       }
     },
+    "/info": {
+      "get": {
+        "tags": [
+          "Diag"
+        ],
+        "summary": "Gets the Dotnet Monitor Version (other configuration information may be added in the future).",
+        "operationId": "GetInfo",
+        "responses": {
+          "400": {
+            "$ref": "#/components/responses/BadRequestResponse"
+          },
+          "401": {
+            "$ref": "#/components/responses/UnauthorizedResponse"
```

Comment:
I don't know where we landed, but this doesn't match with the declaration which seems to make this non-authorized. My assumption would be that this endpoint is not authenticated or authorized and would never return 401.

Reply:
As it is, this route is authenticated because it is on the DiagController, which all of its actions are authenticated.

I'm not convinced that this should be unauthenticated. There may be more sensitive information on this endpoint that we wouldn't want arbitrary users to get. I don't think we would necessary put the authentication information on this route either, so it wouldn't be used to determine how to authenticate (avoiding a chicken-and-egg problem). So I think this declaration is correct.

@dotnet/dotnet-monitor, what do others think?

Reply:
Maybe we should consider putting /version as a separate thing that's not authenticated. That way the version information can always be retrieved without having to worry about leaking other data.

Reply:
Just to make sure I'm understanding correctly: would we have a /version endpoint that's unauthenticated (like /metrics), and then we would additionally have a /info endpoint that is authenticated (where we might put additional sensitive information in the future)? I imagine this _shouldn't_ be difficult to do, but I guess I'm not sure if it's overkill to have an endpoint reserved solely for showing the version. I just imagine in the future if there's other generic info we want to share that doesn't need to be authenticated that we may end up creating a plethora of these one-off endpoints.

I'm not sure if this is possible (or considered a good idea), but would it be possible to overload the /info endpoint such that it has different behavior for authenticated/non-authenticated requests? E.g. calling /info when not authenticated would show basic generic information, such as the version, and then calling /info when you are authenticated would provide you with more detailed, potentially 'secure' info (in addition to the generic information). Not sure if there would be too many implementation/security headaches involved in something like this, but I just want a more futureproof solution than lots of one-off endpoints for users to remember.

Reply:
- I think having the same endpoint behave differently for auth and non-auth would be difficult (we'd probably have to write custom authentication which we are moving away from). 

- I think it would be useful to get the version info without auth (since we may want the information for troubleshooting purposes). If we want to be flexible, would it be reasonable to add unauthenticated /info now and then if we do need secure information we add a separate /secureinfo or something like that?